File: z-clients/java/src/main/java/com/svix/diom/apis/AdminAuthRole.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.DiomException;
import com.svix.diom.HttpClient;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;
import com.svix.diom.models.AdminRoleConfigureIn;
import com.svix.diom.models.AdminRoleConfigureOut;
import com.svix.diom.models.AdminRoleDeleteIn;
import com.svix.diom.models.AdminRoleDeleteOut;
import com.svix.diom.models.AdminRoleGetIn;
import com.svix.diom.models.AdminRoleListIn;
import com.svix.diom.models.AdminRoleOut;
import com.svix.diom.models.ListResponseAdminRoleOut;

public class AdminAuthRole {
    private final HttpClient client;

    public AdminAuthRole(HttpClient client) {
        this.client = client;
    }

    /** Create or update a role */
    public AdminRoleConfigureOut configure(
        final AdminRoleConfigureIn adminRoleConfigureIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-role.configure",
            null,
            adminRoleConfigureIn,
            AdminRoleConfigureOut.class
        );
    }

    /** Delete a role */
    public AdminRoleDeleteOut delete(
        final AdminRoleDeleteIn adminRoleDeleteIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-role.delete",
            null,
            adminRoleDeleteIn,
            AdminRoleDeleteOut.class
        );
    }

    /** Get a role by ID */
    public AdminRoleOut get(
        final AdminRoleGetIn adminRoleGetIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-role.get",
            null,
            adminRoleGetIn,
            AdminRoleOut.class
        );
    }

    /** List all roles */
    public ListResponseAdminRoleOut list(
        final AdminRoleListIn adminRoleListIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-role.list",
            null,
            adminRoleListIn,
            ListResponseAdminRoleOut.class
        );
    }

    /** List all roles */
    public ListResponseAdminRoleOut list(
        
    ) throws DiomException {
        return this.list(
            new AdminRoleListIn()
        );
    }
}

File: z-clients/python/diom/apis/admin_auth_role.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    AdminRoleConfigureIn,
    AdminRoleConfigureOut,
    AdminRoleDeleteIn,
    AdminRoleDeleteOut,
    AdminRoleGetIn,
    AdminRoleListIn,
    AdminRoleOut,
    ListResponseAdminRoleOut,
)


class AdminAuthRoleAsync(ApiBase):
    async def configure(
        self,
        admin_role_configure_in: AdminRoleConfigureIn,
    ) -> AdminRoleConfigureOut:
        """Create or update a role"""
        body = admin_role_configure_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-role.configure",
            body=body,
            response_type=AdminRoleConfigureOut,
        )

    async def delete(
        self,
        admin_role_delete_in: AdminRoleDeleteIn,
    ) -> AdminRoleDeleteOut:
        """Delete a role"""
        body = admin_role_delete_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-role.delete",
            body=body,
            response_type=AdminRoleDeleteOut,
        )

    async def get(
        self,
        admin_role_get_in: AdminRoleGetIn,
    ) -> AdminRoleOut:
        """Get a role by ID"""
        body = admin_role_get_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-role.get",
            body=body,
            response_type=AdminRoleOut,
        )

    async def list(
        self,
        admin_role_list_in: AdminRoleListIn = AdminRoleListIn(),
    ) -> ListResponseAdminRoleOut:
        """List all roles"""
        body = admin_role_list_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-role.list",
            body=body,
            response_type=ListResponseAdminRoleOut,
        )


class AdminAuthRole(ApiBase):
    def configure(
        self,
        admin_role_configure_in: AdminRoleConfigureIn,
    ) -> AdminRoleConfigureOut:
        """Create or update a role"""
        body = admin_role_configure_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.admin.auth-role.configure",
            body=body,
            response_type=AdminRoleConfigureOut,
        )

    def delete(
        self,
        admin_role_delete_in: AdminRoleDeleteIn,
    ) -> AdminRoleDeleteOut:
        """Delete a role"""
        body = admin_role_delete_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.admin.auth-role.delete",
            body=body,
            response_type=AdminRoleDeleteOut,
        )

    def get(
        self,
        admin_role_get_in: AdminRoleGetIn,
    ) -> AdminRoleOut:
        """Get a role by ID"""
        body = admin_role_get_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.admin.auth-role.get",
            body=body,
            response_type=AdminRoleOut,
        )

    def list(
        self,
        admin_role_list_in: AdminRoleListIn = AdminRoleListIn(),
    ) -> ListResponseAdminRoleOut:
        """List all roles"""
        body = admin_role_list_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.admin.auth-role.list",
            body=body,
            response_type=ListResponseAdminRoleOut,
        )
